requirements.txt

```diff
@@ -1,8 +1,11 @@
-requests==2.24.0
-pytest==6.0.1
-pytest-dependency==0.5.1
-pytest-xdist==1.34.0
-apipkg==1.5
-pyyaml==5.4
-zipp==3.1.0
-importlib_metadata==1.7.0
+requests>=2.28.1, <3.0.0
```

Comment:
What would happen if we simply don't put the upper limit and take in all updates automatically? Considering  how short the list is here, wouldn't that be a better approach?

Or to spin it the other way - our goal is to detect breakages early, so that they can be addressed in the regular update cycle (or at least noticed and the work scheduled). If we put the upper limit, we will be blind to breaking changes in future major versions. What is our plan to ensure that gap won't be too large and we get notified?

Reply:
Found the documented procedure in the other PR.

However it might still be good for us to discuss here which approach we are choosing and why.

Reply:
I think it's safer for us to include the upper bounds, which are currently set to just major release updates. While this prevents us from bring in updates, it also protects our users from being the first ones to experience this.  If we have our own process, which we can iterate on (e.g. frequency) in the other PR, then we can bring in these changes at our own timing without negatively impacting non-maintainer contributors to nighthawk.

However, I'm not sure what you mean by `blind to breaking changes in future major versions`. Could you clarify what kind of scenarios you're worried about? I'm just not sure I'm answering your concern.

Reply:
By `blind to breaking changes in future major versions` I meant we won't know whether there are any breaking changes in the new versions until we do try to run the update process.

You did answer my question, thank you and I agree with your reasoning. Let's just make sure the update process is executed regularly at some frequency. We can address this in #884.